### PR TITLE
feature: redirects

### PR DESCRIPTION
**Pull request by lordlabuckdas**
### Major changes

Pages can now redirect, similar to the actual website.

`meta.json` structure:

```json
{
    "/":
        {
            "redirect": "/new/home/"
        },
    "/new/home/":
        {
            "hash": "fb4b89f45c474f455a089130db94b00a",
            "headers": [{"Content-Type": "text/html"}]
        }
}
```

### Minor changes

* Replaced `counter` with `len(visited_urls)`
* `Transfer-Encoding` header has been added to the ignore list

### CODE REVIEW

File: bin/clone [hunk 1]
```diff
@@ -66,13 +66,15 @@ def main():
         loop.run_until_complete(cloner.run())
     except KeyboardInterrupt:
         print_color("\nKeyboardInterrupt received... Quitting", "ERROR")
+    except AssertionError:
```
Comment (afeena) [inline bin/clone:69]: for which assert statement this exception is intended?

Comment (lordlabuckdas) [reply bin/clone:69]: Sometimes when cloner is being run, the connection is closed abruptly by the website - probably due to spam or DDoS protection.

`AssertionError` is raised when the port is not freed up.

```
Traceback (most recent call last):
  File "/home/user/.virtualenvs/snare_venv/bin/clone", line 4, in <module>
    __import__('pkg_resources').run_script('Snare==0.3.0', 'clone')
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/pkg_resources/__init__.py", line 651, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1455, in run_script
    exec(script_code, namespace, namespace)
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/Snare-0.3.0-py3.9.egg/EGG-INFO/scripts/clone", line 92, in <module>
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/Snare-0.3.0-py3.9.egg/EGG-INFO/scripts/clone", line 66, in main
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/Snare-0.3.0-py3.9.egg/snare/cloner.py", line 276, in run
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/Snare-0.3.0-py3.9.egg/snare/cloner.py", line 165, in get_body
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/Snare-0.3.0-py3.9.egg/snare/cloner.py", line 209, in fetch_data
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/aiohttp/client.py", line 520, in _request
    conn = await self._connector.connect(
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/aiohttp/connector.py", line 535, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/aiohttp/connector.py", line 892, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/home/user/.virtualenvs/snare_venv/lib/python3.9/site-packages/aiohttp/connector.py", line 991, in _create_direct_connection
    assert port is not None
AssertionError

```

Comment (afeena) [reply bin/clone:69]: then probably it would be better to move this exception close to the method which raises it

**Review by mzfr — APPROVED** [review]

**Comment by mzfr** [comment]
@afeena The changes look good to me. If @lordlabuckdas isn't adding anything else to this PR then I think we can merge this.

**Pull request merged**